### PR TITLE
Implemented more CompleteRelation API methods for ease of use

### PR DESCRIPTION
**Pull request by lucaspcram**
### Description:
Added a `withRemovedMember` method to CompleteRelation. Instead of using clunky `stream().filter().collect()` calls, you can now do something like:
```java
final CompleteRelation completeRelation = CompleteRelation.shallowFrom(atlas.relation(1L))
.withMembers(atlas.relation(1L).members())
.withRemovedMember(atlas.point(1L));
```

### Potential Impact:
Downstream code that is using `FeatureChange` to remove relation members can be much simpler now.

### Unit Test Approach:
Added some unit tests to ensure this works.

### Test Results:
Tests OK.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java [hunk 1]
```diff
@@ -333,50 +328,85 @@ public int hashCode()
...
     public Iterator<RelationBeanItem> iterator()
     {
-        final List<RelationBeanItem> result = new ArrayList<>();
-        for (int index = 0; index < this.memberIdentifiers.size(); index++)
-        {
-            result.add(new RelationBeanItem(this.memberIdentifiers.get(index),
-                    this.memberRoles.get(index), this.memberTypes.get(index)));
-        }
-        return result.iterator();
+        return this.beanItems.iterator();
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java]: 👍

File: src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java [hunk 2]
```diff
@@ -333,50 +328,85 @@ public int hashCode()
...
+
+        if (firstMatchingIndex != -1)
+        {
+            final String removedRole = this.beanItems.get(firstMatchingIndex).getRole();
+            this.beanItems.remove(firstMatchingIndex);
+            return Optional.of(removedRole);
+        }
+
+        return Optional.empty();
+    }
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java]: :tada

File: src/main/java/org/openstreetmap/atlas/geography/atlas/complete/CompleteArea.java [hunk 3]
```diff
@@ -299,4 +306,13 @@ public CompleteArea withRelations(final Set<Relation> relations)
                 .collect(Collectors.toSet());
         return this;
     }
+
+    @Override
+    public CompleteArea withRemovedRelationIdentifier(final Long relationIdentifier)
+    {
+        this.relationIdentifiers = this.relationIdentifiers.stream()
+                .filter(keepId -> keepId != relationIdentifier.longValue())
+                .collect(Collectors.toSet());
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/complete/CompleteArea.java]: 💯  omg

File: src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java [hunk 4]
```diff
@@ -333,50 +328,85 @@ public int hashCode()
...
+     * If an item is actually removed, this method will return its role wrapped in an
+     * {@link Optional}. If nothing was removed, then the {@link Optional} will be empty.
+     * 
+     * @param identifier
+     *            the id to remove
+     * @param itemType
+     *            the type to remove
+     * @return an {@link Optional} wrapping the role of the removed item
+     */
+    public Optional<String> removeItem(final Long identifier, final ItemType itemType)
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java]: Shouldn't it remove all the roles matching the same item? I know it is uncommon, but sometimes it happens to have the same object with different roles

Comment (lucaspcram) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java]: Good point. My initial thought was to leave it up to the user to deal with this. If they want to remove a specific role, then they should be using the `removeItem` that allows role specification. I can change the behavior here to do what you said, and then change the name to `removeAllMatchingItems` for clarity.

**Pull request merged**